### ISSUE TITLE
Adding memory

### ISSUE THREAD

**Issue by SamPink** [saoudrizwan/claude-dev#2]
First of all this is amazing! 

one thing that would be good though is if it had a chat history, i say this because i keep losing the task when im creating something. 

Similar to this, it would also be good for this to work on existing projects

**Comment by saoudrizwan** [comment]
> one thing that would be good though is if it had a chat history, i say this because i keep losing the task when im creating something.

Task history is on the roadmap!

> Similar to this, it would also be good for this to work on existing projects

You can open Claude Dev in an existing workspace, and it will attempt to complete its tasks within that directory. First it will request permission to view the directory's contents and after examining filenames and extensions, it will determine the type of project (language, frameworks, conventions, etc.) It may also request access to key files, such as the manifest (e.g., package.json), to gain further insight.

However there's a lot of room for improvement here, and I'm currently looking into smarter approaches to analyzing complex projects i.e. through semantic elements within your source code, like symbols and doc comments.

Stay tuned for future updates, and thank you for your feedback!

**Comment by aizenshtat** [comment]
this VSCode extension is unreal! thanks for your work! I like how easy it is to install it.  Do you have the roadmap outlined somewhere? I hope it will become something bigger than just a submission for the Claude competition. LMK if I can help contribute.

**Comment by saoudrizwan** [comment]
@aizenshtat thank you for the kind words! I'll update the README at some point with upcoming features, right now the "roadmap" is just a todo list of fixing bugs/implementing the most requested features. And yes I'd love your help, I will let you know once the repo is in a better state to accept PRs.

**Comment by DaveSanders** [comment]
I ran into this issue today when I decided to move the claude dev panel to the other side of VSC and lost my chat history. 

One trick I'm trying this morning is that I'm using an instructions.md file to maintain the project info and instructions, and at the top of that file I told claude: "As we work through this project, there is a file called progress.md that you have access to. I want you to record all current tasks, past decisions, and next steps in this file as we go along, that way we can pick back up if something happens and I need to restart the tools."

So far, that seems to be working - at least, I'll be able to tell it to review these two MD files when I need to restart a session and maybe give it a leg up.
